File: duplicate-assert/python-unittest/NPathComplexityCheckTest_testVisitTokenBeforeExpressionRange.py
import unittest

class NPathComplexityCheckTest(unittest.TestCase):

    def test_visit_token_before_expression_range(self):
        ast_if = mock_ast(TokenTypes.LITERAL_IF, "if", "mockfile", 2, 2)
        ast_if_left_paren = mock_ast(TokenTypes.LPAREN, "(", "mockfile", 3, 3)
        ast_if.add_child(ast_if_left_paren)
        ast_if_true = mock_ast(TokenTypes.LITERAL_TRUE, "true", "mockfile", 3, 3)
        ast_if.add_child(ast_if_true)
        ast_if_right_paren = mock_ast(TokenTypes.RPAREN, ")", "mockfile", 4, 4)
        ast_if.add_child(ast_if_right_paren)

        ast_ternary = mock_ast(TokenTypes.QUESTION, "?", "mockfile", 1, 1)
        ast_ternary_true = mock_ast(TokenTypes.LITERAL_TRUE, "true", "mockfile", 1, 2)
        ast_ternary.add_child(ast_ternary_true)

        npath_complexity_check_obj = NPathComplexityCheck()

        npath_complexity_check_obj.visit_token(ast_if)
        messages1 = npath_complexity_check_obj.get_messages()

        self.assertEqual(0, len(messages1), "No exception messages expected")

        npath_complexity_check_obj.visit_token(ast_ternary)
        messages2 = npath_complexity_check_obj.get_messages()

        self.assertEqual(0, len(messages2), "No exception messages expected")

File: duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/NPathComplexityCheckTest_testVisitTokenBeforeExpressionRange.cs
using Xunit;

public class NPathComplexityCheckTest : AbstractModuleTestSupport
{
    [Fact]
    public void TestVisitTokenBeforeExpressionRange()
    {
        var astIf = MockAST(TokenTypes.LiteralIf, "if", "mockfile", 2, 2);
        var astIfLeftParen = MockAST(TokenTypes.LParen, "(", "mockfile", 3, 3);
        astIf.AddChild(astIfLeftParen);
        var astIfTrue = MockAST(TokenTypes.LiteralTrue, "true", "mockfile", 3, 3);
        astIf.AddChild(astIfTrue);
        var astIfRightParen = MockAST(TokenTypes.RParen, ")", "mockfile", 4, 4);
        astIf.AddChild(astIfRightParen);

        var astTernary = MockAST(TokenTypes.Question, "?", "mockfile", 1, 1);
        var astTernaryTrue = MockAST(TokenTypes.LiteralTrue, "true", "mockfile", 1, 2);
        astTernary.AddChild(astTernaryTrue);

        var npathComplexityCheckObj = new NPathComplexityCheck();

        npathComplexityCheckObj.VisitToken(astIf);
        var messages1 = npathComplexityCheckObj.GetMessages();

        Assert.Equal(0, messages1.Count, "No exception messages expected");

        npathComplexityCheckObj.VisitToken(astTernary);
        var messages2 = npathComplexityCheckObj.GetMessages();

        Assert.Equal(0, messages2.Count, "No exception messages expected");
    }
}
